include/triton/Analysis/Membar.h

```diff
@@ -10,30 +10,38 @@ namespace mlir {
 
 class OpBuilder;
 
+/// Callback to allow backend to provide more information on whether a barrier
+/// is needed between two operations. Even though two operations access the same
+/// shared memory thay may not require a barrier in between them.
+using MemBarFilterFn = std::function<bool(Operation *, Operation *)>;
```

Comment:
Might be better to change all "membar" to "memBar" and "MemBar" to be more consistent in variable names.

Reply:
sounds good

Reply:
actually this is a bit of a mess as files are named membar as well as commmand lines. I change `MemBarFilterFn` -> `MembarFilterFn` instead. I can refactor it in a separate PR if you think it is important